src/chocolatey/infrastructure.app/services/TemplateService.cs

```diff
@@ -188,5 +196,162 @@ public void generate_file_from_template(ChocolateyConfiguration configuration, T
+                    }
+
+                    this.Log().Info(configuration.RegularOutput ? "{0} Custom templates found at {1}{2}".format_with(templateDirList.Count(), ApplicationParameters.TemplatesLocation, Environment.NewLine) : string.Empty);
+                }
+                else
+                {
+                    this.Log().Info(configuration.RegularOutput ? "No custom templates installed in {0}{2}".format_with(ApplicationParameters.TemplatesLocation, Environment.NewLine) : string.Empty);
+                }
+
+                list_built_in_template_info(configuration, isBuiltInTemplateOverriden, isBuiltInOrDefaultTemplateDefault);
+            }
+            else
+            {
+                if (templateDirList.Contains(_fileSystem.combine_paths(ApplicationParameters.TemplatesLocation, configuration.TemplateCommand.Name)))
+                {
+                    list_custom_template_info(configuration, packageManager);
+                    if (configuration.TemplateCommand.Name == _builtInTemplateName || configuration.TemplateCommand.Name == _builtInTemplateOverrideName)
+                    {
+                        list_built_in_template_info(configuration, isBuiltInTemplateOverriden, isBuiltInOrDefaultTemplateDefault);
+                    }
+                }
+                else
+                {
+                    if (configuration.TemplateCommand.Name.ToLowerInvariant() == _builtInTemplateName || configuration.TemplateCommand.Name.ToLowerInvariant() == _builtInTemplateOverrideName)
+                    {
+                        // We know that the template is not overriden since the template directory was checked
+                        list_built_in_template_info(configuration, isBuiltInTemplateOverriden, isBuiltInOrDefaultTemplateDefault);
+                    }
+                    else
+                    {
+                        throw new ApplicationException("Unable to find requested template '{0}'".format_with(configuration.TemplateCommand.Name));
+                    }
+                }
+            }
+        }
+
+        protected void list_custom_template_info(ChocolateyConfiguration configuration, IPackageManager packageManager)
+        {
+
+            var pkg = packageManager.LocalRepository.FindPackage("{0}.template".format_with(configuration.TemplateCommand.Name));
+            var pkgVersion = (pkg == null) ? "0.0.0" : pkg.Version.to_string();
+            var pkgTitle = (pkg == null) ? "{0} (Unmanaged)".format_with(configuration.TemplateCommand.Name) : pkg.Title;
+            var pkgSummary = (pkg == null) ? 
+                string.Empty : (pkg.Summary != null && !string.IsNullOrWhiteSpace(pkg.Summary.to_string()) ? "{0}".format_with(pkg.Summary.escape_curly_braces().to_string()) : string.Empty);
+            var pkgDescription = (pkg == null) ? string.Empty : pkg.Description.escape_curly_braces().Replace("\n    ", "\n").Replace("\n", "\n  ");

```

Comment:
```suggestion
if (pkg == null)
            {
                this.Log().Error("Unable to find a package for requested template name: {0}".format_with(configuration.TemplateCommand.Name));
                return;
            }

            var pkgVersion = pkg.Version.to_string();
            var pkgTitle = pkg.Title;
            var pkgSummary = pkg.Summary != null && !string.IsNullOrWhiteSpace(pkg.Summary.to_string()) ? "{0}".format_with(pkg.Summary.escape_curly_braces().to_string()) : string.Empty;
            var pkgDescription = pkg.Description.escape_curly_braces().Replace("\n    ", "\n").Replace("\n", "\n  ");
```

Reply:
I don't this this would work for templates that are installed manually instead of via `.template` package

Reply:
Ah, I hadn't thought of that side of things, I was only thinking about templates installed via a package.  In which case, I think we need some comments around this section to make it clear why we are looking to handle both ways of doing it.

And the suggestion would change to doing a single `if (pkg != null)` check, rather than doing it multiple times.

Reply:
> And the suggestion would change to doing a single if (pkg != null) check, rather than doing it multiple times.

Done, did it via a bool, as if I did it inside an if, there would be variable scoping issues, or lots of mess declaring variables beforehand.